### PR TITLE
Install dev dependencies with poetry

### PR DESCRIPTION
**Pull request by gsingh93**
Add all dev dependencies to `pyproject.toml` and remove `dev-requirements.txt`. `setup-dev.sh` now installs poetry if it's not already installed and then installs all dependencies with `poetry install  --with dev`.

We should eventually install normal dependencies with poetry instead of pip, but let's start with just the dev ones and see if everything goes fine.

### CODE REVIEW

**Comment by gsingh93** [comment]
@patryk4815 Could you look into the lock file issue?

**Comment by patryk4815** [comment]
@gsingh93 fix:
```diff
diff --git a/nix/pwndbg.nix b/nix/pwndbg.nix
index 2bb0b5fb..5801d7e1 100644
--- a/nix/pwndbg.nix
+++ b/nix/pwndbg.nix
@@ -13,6 +13,8 @@ let
   ]);

   pyEnv = pkgs.poetry2nix.mkPoetryEnv {
+    groups = [];  # put [ "dev" ] to build "dev" dependencies
+    checkGroups = [];  # put [ "dev" ] to build "dev" dependencies
     projectDir = inputs.pwndbg;
     python = python3;
     overrides = pkgs.poetry2nix.overrides.withDefaults (self: super: {
```

> groups: Which Poetry 1.2.0+ dependency groups to install (default: [ "dev" ]).
> checkGroups: Which Poetry 1.2.0+ dependency groups to run unit tests (default: [  "dev" ]).

File: setup-dev.sh [hunk 1]
```diff
@@ -188,6 +188,12 @@ if linux; then
         PWNDBG_VENV_PATH="./.venv"
     fi
     echo "Using virtualenv from path: ${PWNDBG_VENV_PATH}"
-    PYTHON=${PWNDBG_VENV_PATH}/bin/python
-    ${PYTHON} -m pip install -r dev-requirements.txt
+
+    # Install poetry if not already installed
+    if ! hash poetry 2> /dev/null; then
+        curl -sSL https://install.python-poetry.org | python3 -
```
Comment (disconnect3d) [inline setup-dev.sh:194]: Shouldn't this be executed by the venv python?

Comment (gsingh93) [reply setup-dev.sh:194]: No, the poetry docs explicitly say it shouldn't be installed in the project venv.

**Comment by gsingh93** [comment]
Ping :)

**Pull request merged**